pkg/madmin/service-commands.go

```
@@ -18,69 +18,79 @@
+type ServerVersion struct {
+	Version  string `json:"version"`
+	CommitID string `json:"commitID"`
 }
 
-// ServiceStatus - Connect to a minio server and call Service Status Management API
-// to fetch server's storage information represented by ServiceStatusMetadata structure
-func (adm *AdminClient) ServiceStatus() (ServiceStatusMetadata, error) {
-
-	// Prepare web service request
-	reqData := requestData{}
-	reqData.queryValues = make(url.Values)
-	reqData.queryValues.Set("service", "")
-	reqData.customHeaders = make(http.Header)
-	reqData.customHeaders.Set(minioAdminOpHeader, "status")
+// ServiceStatus - contains the response of service status API
+type ServiceStatus struct {
+	ServerVersion ServerVersion `json:"serverVersion"`
+	Uptime        time.Duration `json:"uptime"`
+}
 
-	// Execute GET on bucket to list objects.
-	resp, err := adm.executeMethod("GET", reqData)
+// ServiceStatus - Connect to a minio server and call Service Status
+// Management API to fetch server's storage information represented by
+// ServiceStatusMetadata structure
+func (adm *AdminClient) ServiceStatus() (ss ServiceStatus, err error) {
+	// Request API to GET service status
+	resp, err := adm.executeMethod("GET", requestData{relPath: "/v1/service"})
 	defer closeResponse(resp)
 	if err != nil {
-		return ServiceStatusMetadata{}, err
+		return ss, err
 	}
 
 	// Check response http status code
 	if resp.StatusCode != http.StatusOK {
-		return ServiceStatusMetadata{}, httpRespToErrorResponse(resp)
+		return ss, httpRespToErrorResponse(resp)
 	}
 
-	// Unmarshal the server's json response
-	var serviceStatus ServiceStatusMetadata
-
 	respBytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return ServiceStatusMetadata{}, err
+		return ss, err
 	}
 
-	err = json.Unmarshal(respBytes, &serviceStatus)
-	if err != nil {
-		return ServiceStatusMetadata{}, err
-	}
-
-	return serviceStatus, nil
+	err = json.Unmarshal(respBytes, &ss)
+	return ss, nil
```

Error is ignored? for json.Unmarshal?

---

Fixed.